pkgs/build-support/cc-wrapper/default.nix

```diff
@@ -245,12 +245,19 @@ stdenv.mkDerivation {
       wrap ${targetPrefix}gdc $wrapper $ccPath/${targetPrefix}gdc
     ''
 
-    + optionalString cc.langFortran or false ''
+    + optionalString cc.langFortran or false (''
       wrap ${targetPrefix}gfortran $wrapper $ccPath/${targetPrefix}gfortran
       ln -sv ${targetPrefix}gfortran $out/bin/${targetPrefix}g77
       ln -sv ${targetPrefix}gfortran $out/bin/${targetPrefix}f77
       export named_fc=${targetPrefix}gfortran
     ''
+    # Darwin aarch64 fortran compilations seem to fail otherwise, see:
+    # https://github.com/NixOS/nixpkgs/issues/140041
+    + (if (stdenvNoCC.isDarwin && stdenvNoCC.isAarch64) then ''
+      export fortran_hardening="pic strictoverflow relro bindnow"
```

Comment:
Can we use a consistent naming for this?
https://github.com/NixOS/nixpkgs/blob/23b08bae5c508f7625b737446b65a590a5a722c6/pkgs/build-support/cc-wrapper/add-hardening.sh#L13

Reply:
> Can we use a consistent naming for this?
> 
> https://github.com/NixOS/nixpkgs/blob/23b08bae5c508f7625b737446b65a590a5a722c6/pkgs/build-support/cc-wrapper/add-hardening.sh#L13

But the `strictprotector` hardening flag is not an "unsupported" flag, it's just causes a certain issue. Also I think the variable name should be distinguished from the variable "hardening_unsupported_flags" because it's a list specific to fortran.